### PR TITLE
Feature/improve demo app

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
# Changes

Upgrades the demo app and e2e test setup. This PR contains multiple changes:
- Some dependencies upgrades
- Improves the app used for tests
- Improves browser-based tests
  - Migrates them from testcafé to playwright
  - improves setup and teardown by creating a public and private file and removing them from the Pod after the tests
  - Re-enables browser-based tests running against dev-next
- Improves how CI runs these tests
- Fixes codesandbox integration to enforce it only deploys the browser codebase

# Checklist

- [X] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [ ] ~The changelog has been updated, if applicable.~
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Review by NSeydoux — APPROVED** [review]
Some improvements may be added later on based on implementation feedback in other repos, but this is already a massive improvement over what we had before.

File: .github/workflows/e2e-browser.yml [hunk 1]
```diff
@@ -28,40 +32,38 @@ jobs:
...
           key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
       - run: npm ci
       - run: npm run bootstrap -- --ci
       - run: npm run build
+      - name: Install e2e tests dependencies
+        run: |
+          cd e2e/browser/
+          npm ci
+          npx playwright install-deps chromium firefox webkit
+          npx playwright install
```
Comment (ThisIsMissEm) [inline .github/workflows/e2e-browser.yml:46]:
```suggestion
          npx playwright install --with-deps
```

This does the same as the above, but with a single step.

File: e2e/browser/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
```
Comment (ThisIsMissEm) [inline e2e/browser/.eslintrc.js:1]: I think this change in our linting configuration means we don't need this anymore: https://github.com/inrupt/javascript-style-configs/pull/22

File: e2e/browser/.gitignore [hunk 3]
```diff
@@ -1,5 +1,4 @@
-# TestCafe test failures can record screenshots, which can be placed here.
-e2e-browser-failures/
+# Ignore the playwright test result recordings:
+test-results/
 
-# TestCafe can record requests (and responses), but we don't want those in Git.
-testcafe-requests
\ No newline at end of file
+.env
```
Comment (ThisIsMissEm) [inline e2e/browser/.gitignore:4]:
```suggestion
.env
.env*.local
```

That way the `.env.local` and `.env.test.local` files will also be git ignored.

Comment (ThisIsMissEm) [reply e2e/browser/.gitignore:4]: Actually, we can just remove this line as it's `../env/*` now that configures things.
```suggestion
```

File: e2e/browser/README.md [hunk 4]
```diff
@@ -10,70 +10,46 @@ The commands below assume they are being executed in this directory, and that
...
+- how the tested application is deployed (command run, port...)
+
+You may need to install some components for playwright to be able to run the 
+specified browsers:
+
+```script
+cd e2e/browser/
+npm ci
+npx playwright install-deps chromium firefox webkit
+npx playwright install
```
Comment (ThisIsMissEm) [inline e2e/browser/README.md:34]:
```suggestion
npx playwright install --with-deps
```

File: e2e/browser/pages/AppPage.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    await this.page.fill('[data-testid="idp_input"]', url);
+
+    // Click the login button:
+    await this.page.click('[data-testid="idp_login_button"]');
+    await this.page.waitForNavigation({
+      waitUntil: "networkidle",
+    });
+  }
+
+  async fetchResource(url: string) {
```
Comment (ThisIsMissEm) [inline e2e/browser/pages/AppPage.ts:46]: Might be good to move both `fetchResource` and `getFetchResponse` and related code into a `TestPage` Page, so those are isolated from the setup of logging in?

File: .github/workflows/e2e-browser.yml (hunk 1 above)
Comment (NSeydoux) [reply .github/workflows/e2e-browser.yml]: Fixed in 068a5b49

File: .github/workflows/e2e-browser.yml [hunk 6]
```diff
@@ -28,40 +32,38 @@ jobs:
...
-            node_modules
+          path: node_modules
             packages/*/node_modules
           key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
       - run: npm ci
       - run: npm run bootstrap -- --ci
       - run: npm run build
+      - name: Install e2e tests dependencies
+        run: |
+          cd e2e/browser/
```
[the remark this answers is not in the document]
Comment (NSeydoux) [reply .github/workflows/e2e-browser.yml]: Fixed in 068a5b49

File: e2e/browser/.eslintrc.js (hunk 2 above)
Comment (NSeydoux) [reply e2e/browser/.eslintrc.js:1]: It looks like the whole e2e folder isn't linted at all. Will fix the config and the associated errors in a separate PR.

**Pull request merged**